### PR TITLE
Data Reformat

### PR DESCRIPTION
**Pull request by Magisus**
Added a module to reformat set data from MTG JSON into data indexed and sorted by card name. Cards include set data as lists of the sets they were printed in.

Chose single quoted strings as default for this project.

### CODE REVIEW

File: spec/data_formatter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          'RTR' => {
+            'name'  => 'Return to Ravnica',
+            'code'  => 'RTR',
+            'cards' => [{'name' => 'Forest', 'type' => 'Land', 'id' => '10'},
+                        {'name' => 'Simic Charm', 'type' => 'Instant'}]
+          }
+        }
+      }
+
+      it 'should add set data to card data' do
```
Comment (connorlay) [inline spec/data_formatter_spec.rb]: For spec descriptions avoid 'should' or 'is expected to'. Use an active voice to describe the behavior under test.

~~~ruby
it 'adds set data to card data' do
...
end
~~~

File: spec/data_formatter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        cards = DataFormatter.reformat(data)
+        expect(cards['Forest']).to eq({'name' => 'Forest',
+                                       'type' => 'Land',
+                                       'setName' => ['Nemesis',
+                                                     'Return to Ravnica'],
+                                       'setCode' => ['NMS', 'RTR']})
+        expect(cards['Forest']).not_to include('id')
+      end
+
+      it 'should sort data by card name' do
```
Comment (connorlay) [inline spec/data_formatter_spec.rb]: Same as above

File: lib/mtg_cli/data_formatter.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        reformatted_cards.merge!(parse_set(data)) do |key, old_card, new_card|
+          old_card['setName'].push(new_card['setName'].first)
+          old_card['setCode'].push(new_card['setCode'].first)
+          old_card
+        end
+      end
+      reformatted_cards.sort_by { |name, data| name }.to_h
+    end
+
+    def parse_set(set_hash)
```
Comment (connorlay) [inline lib/mtg_cli/data_formatter.rb]: This is a pretty big method that's hard to parse... is there any way you could split it up into smaller sub methods?

**Comment by Magisus** [comment]
Yeah those comments make a lot of sense now that I look at it. Check it out now.

**Comment by connorlay** [comment]
Looks good to me!
